src/compiletest/runtest.rs

```diff
@@ -1013,8 +1013,8 @@ fn check_expected_errors(revision: Option<&str>,
         expected_errors.iter()
                         .fold((false, false),
```

Comment:
Given this code block:

```rust
fn main() {
  let i = 1; //~ NOTE: some note
  let j = 2; //~ HELP: some help 
}
```

The current implementation checks for `note` and `help` in one pass through. I'm not sure how it'd be possible using `any` unless we were expecting **both** `help` and `note` on the same line. I agree it's not clean, and I'm willing to clean it up, but would rather do that in a different PR.